_tools/generateFontData.js

```diff
@@ -23,32 +23,45 @@ const {
 
 const srcDirectory = path.resolve(__dirname, "../", "src");
 const fontsDirectory = path.resolve(srcDirectory, "fonts");
-const dataDirectory = path.resolve(srcDirectory, "_data");
-const fontsStylesheetPath = path.resolve(srcDirectory, "css", "font.css");
-const fontJsPath = path.resolve(srcDirectory, "js", "font.js");
+const dataDirectory = path.resolve(srcDirectory, "_data/fonts");
+const fontsStylesheetPath = path.resolve(srcDirectory, "css", "fonts.css");
+const fontJsPath = path.resolve(srcDirectory, "js", "fonts.js");
+
+const slugify = text => {
+	return text
+		.toLowerCase()
+		.replace(/ /g, "-")
+		.replace(/[-]+/g, "-")
+		.replace(/[^\w-]+/g, "");
+};
 
 const assert = (condition, message) => {
 	if (!condition) {
 		throw new Error(message);
 	}
 };
 
+const _appendFile = util.promisify(fs.appendFile);
```

Comment:
I'm not a fan of mutating things. I'd rather build up the whole result and write it at once, instead of appending things. Many things can go wrong with appending, you need to handle order etc. What do you think?

Reply:
Hmm. I'll see if I can refactor this.